src/datapoint.py

```diff
@@ -0,0 +1,114 @@
+from typing import Dict
+import numpy as np
+from enum import Enum
+from abc import ABC
+
+
+class Method(Enum):
+    C = 'Classification'
+    R = 'Regression'
```

Comment:
This is very not explicit and will not be intuitive to use.
```suggestion
    CLASSIFICATION = 'Classification'
    REGRESSION = 'Regression'
```